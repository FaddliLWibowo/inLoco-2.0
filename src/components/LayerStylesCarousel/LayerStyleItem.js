import React from 'react'

const LayerStyleItem = ({layer, style, index, onStyleClick, onStyleMouseOver, onStyleMouseOut}) => {

    let itemClassName = 'layer-styles-carousel--list-item'
    if (layer && layer.selectedLayerStyleId === index) {
        itemClassName += ' selected'
    }

    function styleClick() {
        return onStyleClick(layer, index)
    }

    function styleMouseOver() {
        return onStyleMouseOver(layer, index)
    }

    function styleMouseOut() {
        return onStyleMouseOut(layer, index)
    }

    return (
        <li
            className={itemClassName}
            onClick={styleClick}
        >
            { style ?
                <img
                    className="layer-styles-carousel--image"
                    src={style.thumb}
                    alt={style.title}
                    onMouseOver={styleMouseOver}
                    onMouseOut={styleMouseOut}
                />
            : '' }
        </li>
    )
}

export default LayerStyleItem
