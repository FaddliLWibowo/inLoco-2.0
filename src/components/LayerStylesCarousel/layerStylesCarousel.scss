.layer-styles-carousel {
    position: relative;
    margin-bottom: 10px;

    &::selection {
        background: none;
    }

    &--arrow {
        z-index: 2;
        position: absolute;
        top: 33%;
        padding: 7px;
        width: 25px;
        height: 25px;
        font-size: 12px;
        text-align: center;
        color: $layer-button-carousel-text-color;
        border-radius: 50%;
        background-color: $layer-button-carousel-bg-color;
        cursor: pointer;

        &.remove {
            display: none;
        }

        &:hover,
        &:focus {
            color: $layer-button-carousel-text-color-hover;
            background-color: $layer-button-carousel-bg-color-hover;
        }

        &.fa-chevron-left:before { margin-left: -3px; }
        &.fa-chevron-right:before { margin-right: -2px; }
    }
    .layer-styles-carousel--arrow:last-child{ right: 16px; }

    &--list-container {
        display: inline-block;
        margin-top: 10px;
        margin-left: 2px;
        overflow: hidden;
        width: 328px;
    }

    &--list {
        height: 39px;
        position: relative;
        transition-duration: .2s;
    }

    &--list-item {
        display: inline-block;
        height: 39px;
        margin: 0 2px;
        border: 1px solid rgba(0,0,0,0); // every item has a 1px transparent border

        &.selected {
            background-color: $layer-styles-list-item-selected-background;

            .layer-styles-carousel--image {
                opacity: .5;
            }

        }
    }

    &--image {
        border: 1px solid $layer-list-secondary-text-color;
        cursor: pointer;
        height: 37px;
        width: 60px;

        &:hover {
            border: 1px solid $layer-list-area-title-color;
        }

        &::selection {
            background: none;
        }
    }
}
