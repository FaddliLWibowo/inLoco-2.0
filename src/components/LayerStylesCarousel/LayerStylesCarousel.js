import React from 'react'
import LayerStyleItem from './LayerStyleItem';

const LayerStylesCarousel = ({ layer, onArrowLeftClick, onArrowRightClick, onStyleClick }) => {

    const STYLE_WIDTH = 66 //magic number

    let selectedLayerId

    let leftArrowClassName  = 'layer-styles-carousel--arrow fa fa-chevron-left'
    let rightArrowClassName = 'layer-styles-carousel--arrow fa fa-chevron-right'

    if (layer.styles.length < 6) {
        leftArrowClassName  += ' hidden'
        rightArrowClassName += ' hidden'
    } else {
        if (!layer.stylesPositionCounter) {
            leftArrowClassName += ' hidden'
        } else if (layer.stylesPositionCounter === layer.styles.length - 5) {
            rightArrowClassName += ' hidden'
        }
    }

    let carouselListStyle = {
        width: (layer.styles.length * STYLE_WIDTH) + 'px',
        marginLeft: -( (layer.stylesPositionCounter || 0) * STYLE_WIDTH) + 'px',
    }

    function arrowLeftClick() {
        return onArrowLeftClick(layer)
    }

    function arrowRightClick() {
        return onArrowRightClick(layer)
    }

    return (
        <div className="layer-styles-carousel">
            <a role="button" className={leftArrowClassName} onClick={arrowLeftClick}></a>
            <div className="layer-styles-carousel--list-container">
                <ul className="layer-styles-carousel--list" style={carouselListStyle}>
                    {layer.styles.map((style, indexStyle) => {
                        return (
                            <LayerStyleItem
                                layer={layer}
                                style={style}
                                key={indexStyle}
                                index={indexStyle}
                                onStyleClick={onStyleClick}
                            />
                        )
                    })}
                </ul>
            </div>
            <a role="button" className={rightArrowClassName} onClick={arrowRightClick}></a>
        </div>
    )
}

export default LayerStylesCarousel
