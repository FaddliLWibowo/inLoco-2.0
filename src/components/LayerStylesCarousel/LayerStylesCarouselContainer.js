import React from 'react';
import LayerStylesCarousel from './LayerStylesCarousel';
import { connect } from 'react-redux';
import { slideLeftStyles, slideRightStyles } from '../../actions/actions.js';

const mapStateToProps = (state, ownProps) => {
    return {
        layer: ownProps.layer,
    }
}

const mapDispatchToProps = (dispatch) => {
    return {
        onArrowLeftClick: (item) => {
            dispatch(slideLeftStyles(item));
        },
        onArrowRightClick: (item) => {
            dispatch(slideRightStyles(item));
        },
    }
}

const LayerStylesCarouselContainer = connect(
    mapStateToProps,
    mapDispatchToProps
)(LayerStylesCarousel)

export default LayerStylesCarouselContainer
