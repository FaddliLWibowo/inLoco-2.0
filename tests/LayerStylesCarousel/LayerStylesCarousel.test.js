import React from 'react';
import LayerStylesCarousel from '../../src/components/LayerStylesCarousel/LayerStylesCarousel.js';
import renderer from 'react-test-renderer';

it('LayerStylesCarousel component renders correctly with no data', () => {
    const component = shallow(<LayerStylesCarousel/>);
    expect(component).toMatchSnapshot();
});
