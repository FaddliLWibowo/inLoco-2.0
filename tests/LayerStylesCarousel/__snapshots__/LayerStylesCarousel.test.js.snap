// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LayerStylesCarousel component renders correctly with no data 1`] = `
<div
  className="layer-styles-carousel"
>
  <a
    className="layer-styles-carousel--arrow fa fa-chevron-left"
    onClick={[Function]}
    role="button"
  />
  <div
    className="layer-styles-carousel--list-container"
  >
    <ul
      className="layer-styles-carousel--list"
      style={
        Object {
          "marginLeft": "0px",
          "width": "0px",
        }
      }
    />
  </div>
  <a
    className="layer-styles-carousel--arrow fa fa-chevron-right"
    onClick={[Function]}
    role="button"
  />
</div>
`;
